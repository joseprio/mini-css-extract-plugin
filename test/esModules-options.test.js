import path from 'path';

import webpack from 'webpack';

describe('EsModules', () => {
  it('should generate ES6 modules', (done) => {
    const casesDirectory = path.resolve(__dirname, 'cases');
    const directoryForCase = path.resolve(casesDirectory, 'esModules');
    // eslint-disable-next-line import/no-dynamic-require, global-require
    const webpackConfig = require(path.resolve(
      directoryForCase,
      'webpack.config.js'
    ));
    const compiler = webpack({
      ...webpackConfig,
      mode: 'development',
      context: directoryForCase,
      cache: false,
    });

    compiler.run((err1, stats) => {
      const { modules } = stats.toJson();
      let foundModule = false;
      modules.forEach((module) => {
        if (module.name === './style.css') {
          foundModule = true;
          expect(module.source).toMatchSnapshot();
        }
      });
      expect(foundModule).toBe(true);
      done();
    });
  });
});
