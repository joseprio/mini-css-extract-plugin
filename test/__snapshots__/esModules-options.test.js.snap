// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsModules should generate ES6 modules 1`] = `
"// extracted by mini-css-extract-plugin
export default {\\"test\\":\\"MHBvAShXnAxWEC-jxsBd\\"};"
`;
